packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -670,6 +670,31 @@ class DDSCarousel extends HostListenerMixin(StableSelectorMixin(LitElement)) {
   firstUpdated() {
     this._cleanAndCreateObserverResize({ create: true });
     this._cleanAndCreateObserverIntersection({ create: true });
+    // Reflect the current writing mode in the dir attribute so that we can
+    // make styling adjustments.
+    this.setAttribute(
+      'dir',
+      window.getComputedStyle(this).getPropertyValue('direction')
+    );
+  }
+
+  updated(changedProperties) {
+    if (
+      !changedProperties.has('start') &&
+      !changedProperties.has('_contentsBaseWidth') &&
+      !changedProperties.has('_gap') &&
+      !changedProperties.has('pageSize')
+    ) {
+      return;
+    }
+
+    if (!this._contentsNode) {
+      return;
+    }
+
+    const startPosition =
+      (-this.start * (this._contentsBaseWidth + this._gap)) / this.pageSize;
+    this._contentsNode.style.insetInlineStart = `${startPosition}px`;
```

Comment:
If you're up for some friendly debate, I'm not sure I understand the value of using the return early pattern here. IMHO it makes this a little less readable and more difficult to add more functionality into the updated hook in the future. Not a blocking comment 🙂

Reply:
Always! A couple reasons I like early returns:

* Rather then writing a really long condition, with brackets and potentially a mix or logical OR and AND expressions, or multiple nested statements, I like breaking it up into multiple flat conditions. Totally subjective, but I think this way is easier to read.
* Broken up into multiple early returns, it's easier to document each. I was lazy here, but I've done that in https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11030/commits/64583dfc26d5004c54c97249007895e8e101f9fd
* I like getting the early returns out of the way so the meat of the function is as flat (minimal nesting) as possible.
* Salva Molina indoctrinated me to early returns 😛 

I agree with you about this being difficult to update in the future. I separated this concern into a private method so that `update` has room to grow if needed in future.

Reply:
Love that compromise. Thanks, Matt!